test/new-e2e/pkg/runner/profile.go

```diff
@@ -98,6 +99,24 @@ func newProfile(projectName string, environments []string, store parameters.Stor
 	return p
 }
 
+func defaultEnvironments(environments string, defaultsEnvironments map[string]string) string {
+	defaultedEnvironmentsMap := maps.Clone(defaultsEnvironments)
+	environmentsList := strings.Split(environments, " ")
+	environmentsMap := make(map[string]string)
+	for _, env := range environmentsList {
+		parts := strings.Split(env, "/")
+		if len(parts) == 2 {
+			environmentsMap[parts[0]] = parts[1]
+		}
+	}
+	maps.Copy(defaultedEnvironmentsMap, environmentsMap)
+	defaultEnvironments := ""
+	for k, v := range defaultedEnvironmentsMap {
+		defaultEnvironments += fmt.Sprintf("%s/%s ", k, v)
+	}
+	return strings.TrimSpace(defaultEnvironments)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
I feel like we can skip one map. Also opinionated change to first copy default values and then overriding keys with elements from `environments`
**💬 suggestion**
```suggestion
	environmentsList := strings.Split(environments, " ")
	// set merged map capacity to worst case scenario of no overlapping key
	mergedEnvironmentsMap := make(map[string]string, len(defaultEnvironments) + len(environmentsList))
	maps.Copy(mergedEnvironmentsMap, defaultsEnvironments)
	for _, env := range environmentsList {
		parts := strings.Split(env, "/")
		if len(parts) == 2 {
			mergedEnvironmentsMap[parts[0]] = parts[1]
		}
	}
	maps.Copy(defaultedEnvironmentsMap, environmentsMap)
	mergedEnvironmentsList := make([]string, 0, len(mergedEnvironmentsMap))
	for k, v := range mergedEnvironmentsMap {
		mergedEnvironmentsList = mergedEnvironmentsList.append(mergedEnvironmentsList, fmt.Sprintf("%s/%s ", k, v))
	}
	return strings.Join(mergedEnvironmentsList, " ")
```

Reply:
I should have done a single suggestion 🤦‍♀️

Reply:
That's fine no worries! Applied them and I will fix the remaining things locally now!